framework/test/unit/specs/modules/keys/register_asset.spec.ts

```diff
@@ -0,0 +1,340 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { randomBytes } from 'crypto';
+import { codec } from '@liskhq/lisk-codec';
+import { hash } from '@liskhq/lisk-cryptography';
+import { Transaction, transactionSchema } from '@liskhq/lisk-chain';
+import { RegisterAsset } from '../../../../../src/modules/keys/register_asset';
+import { createFakeDefaultAccount, StateStoreMock } from '../../../../utils/node';
+import * as fixtures from './fixtures.json';
+
+let decodedMultiSignature: any;
+let validTestTransaction: any;
+let multisignatureSender: any;
+let targetMultisigAccount: any;
+let convertedAccount: any;
+let stateStore: any;
+let storeAccountGetStub: jest.SpyInstance;
+let storeAccountSetStub: jest.SpyInstance;
+let registerAsset: RegisterAsset;
+let reducerHandler: any;
+
+describe('register asset', () => {
+	const defualtTestCase = fixtures.testCases[0];
+	beforeEach(() => {
+		registerAsset = new RegisterAsset();
+		const buffer = Buffer.from(defualtTestCase.output.transaction, 'base64');
+		const id = hash(buffer);
+		const decodedBaseTransaction = codec.decode<Transaction>(transactionSchema, buffer);
+		const decodedAsset = codec.decode<any>(registerAsset.assetSchema, decodedBaseTransaction.asset);
+		decodedMultiSignature = {
+			...decodedBaseTransaction,
+			asset: decodedAsset,
+			id,
+		};
+		validTestTransaction = new Transaction(decodedMultiSignature);
+
+		multisignatureSender = createFakeDefaultAccount({
+			address: Buffer.from(defualtTestCase.input.account.address, 'base64'),
+		});
+
+		targetMultisigAccount = createFakeDefaultAccount({
+			address: Buffer.from(defualtTestCase.input.account.address, 'base64'),
+			balance: BigInt('94378900000'),
+		});
+		convertedAccount = createFakeDefaultAccount({
+			address: Buffer.from(defualtTestCase.input.account.address, 'base64'),
+			balance: BigInt('94378900000'),
+			keys: {
+				...validTestTransaction.asset,
+			},
+		});
+
+		stateStore = new StateStoreMock();
+		storeAccountGetStub = jest.spyOn(stateStore.account, 'getOrDefault').mockResolvedValue(
+			createFakeDefaultAccount({
+				address: Buffer.from(defualtTestCase.input.account.address, 'base64'),
+			}) as never,
+		);
+
+		storeAccountGetStub = jest
+			.spyOn(stateStore.account, 'get')
+			.mockResolvedValue(targetMultisigAccount);
+
+		storeAccountSetStub = jest.spyOn(stateStore.account, 'set');
+
+		reducerHandler = {};
+	});
+
+	describe('validateAsset', () => {
```

Comment:
maybe check if the schema is defined/will be validated properly as done in https://github.com/LiskHQ/lisk-sdk/blob/feature/on_chain_architecture/framework/test/unit/specs/modules/dpos/transaction_assets/vote_transaction_asset.spec.ts#L79?

Reply:
Full tests for the schema added to `validateSchema`